python/cudf/cudf/core/dataframe.py

```diff
@@ -260,11 +268,109 @@ def __init__(self, data=None, index=None, columns=None, dtype=None):
         if dtype:
             self._data = self.astype(dtype)._data
 
+    def _init_from_series_list(self, data, columns, index):
+        if index is None:
+            # When `index` is `None`, the final index of
+            # resulting dataframe will be union of
+            # all Series's names.
+            final_index = as_index(_get_union_of_series_names(data))
```

Comment:
I still don't see where we're erroring on duplicate index values, are we?

Reply:
This part of code is the construction of end-result index using list of `Series` objects `name` attrs.

We are actually erroring here for duplicate index values: https://github.com/rapidsai/cudf/pull/5340/files#diff-d4816377962eac0ab58997c5037843afR331

Associated test is https://github.com/rapidsai/cudf/pull/5340/files#diff-7b1f0eb2073ee903516e675d55c4a9e0R6765